server.go

```diff
@@ -106,6 +113,10 @@ func NewServer(opt ...ServerOption) *Server {
 	for _, o := range opt {
 		o(&opts)
 	}
+	if opts.codec == nil {
+		// Set the default codec.
+		opts.codec = &protoCodec{}
```

Comment:
all done.

On Wed, Apr 1, 2015 at 2:28 PM, Qi Zhao <toqizhao@gmail.com> wrote:

>
>
> On Wed, Apr 1, 2015 at 2:22 PM, David Symonds <notifications@github.com>
> wrote:
>
>> In server.go
>> <https://github.com/grpc/grpc-go/pull/146#discussion_r27613201>:
>>
>> > @@ -106,6 +113,10 @@ func NewServer(opt ...ServerOption) *Server {
>> >  	for _, o := range opt {
>> >  		o(&opts)
>> >  	}
>> > +	if opts.codec == nil {
>> > +		// Set the default codec.
>> > +		opts.codec = &protoCodec{}
>>
>> That's not how Go interfaces work. Here, protoCodec is an empty struct.
>> It doesn't need to be a pointer, because the methods for the Codec
>> interface are on protoCodec, not *protoCodec. If a user has a complex
>> codec, they can implement those methods on *myCodec and pass
>> &myCodec{...} to the relevant functions in this package.
>>
> This is definitely understood. I think my concern was just about which way
> can simplify the user's thinking. But perhaps it is not necessary...
>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/grpc/grpc-go/pull/146/files#r27613201>.
>>
>
>